samza-core/src/main/java/org/apache/samza/runtime/RemoteApplicationRunner.java

```diff
@@ -85,7 +91,7 @@ public void kill() {
     // since currently we only support single actual remote job, we can get its status without
     // building the execution plan.
     try {
-      JobConfig jc = new JobConfig(JobPlanner.generateSingleJobConfig(appDesc.getConfig()));
+      JobConfig jc = new JobConfig(JobPlanner.generateSingleJobConfig(config));
```

Comment:
This is a question for `kill`, `status`, and `waitForFinish`:
Should you be calling `JobPlanner.generateSingleJobConfig` even when config loader factory is specified? It seems like you would want to call `JobRunner`, similar to how you changed it for `run`.

Reply:
`JobPlanner.generateSingleJobConfig` is a util method which overrides `job.name` and `job.id` based on `app.name` and `app.id`, which I suppose is safe to keep. 

`kill`, `status` and `waitForFinish` are still using calling JobRunner.

Reply:
But in `run`, you don't call `JobPlanner.generateSingleJobConfig`. For consistency, it seems like you should call `generateSingleJobConfig` for all cases or for no cases.

Reply:
Got you, `JobPlanner.generateSingleJobConfig` is not being used in `run()` in the legacy code path neither, do you think we should explore this separately from this PR?

Reply:
It looks like `ExecutionPlanner.createJobGraph` (part of planning) uses `JobPlanner.generateSingleJobConfig`, so it looks like `JobPlanner.generateSingleJobConfig` is used in `run` in the legacy path. Can you please check if I am reading the code correctly here?

Reply:
Updated.

Reply:
Thanks for pointing this out!